ntoskrnl/ke/i386/kiinit.c

```diff
@@ -676,7 +677,7 @@ KiSystemStartupBootStack(VOID)
 
     /* Initialize the kernel for the current CPU */
     KiInitializeKernel(&KiInitialProcess.Pcb,
-                       (PKTHREAD)KeLoaderBlock->Thread,
+                       (PKTHREAD)__readfsdword(KPCR_CURRENT_THREAD),
```

Comment:
But, is KPCR_CURRENT_THREAD offset in the FS initialized?

Reply:
Actually in my question I'm concerned for any "current CPU", be it the boot processor or the APs.
But yes, indeed after checking, for APs KiInitializePcr is get called during their preparation in the loop in KeStartAllProcessors, so that when each CPU starts the KiSystemStartup entry point, their Pcr contain already the good data.

Reply:
This kind of stuff is why I rewrote the init code on x64. Less "this is how Windows does it", less hacks, more better code :)
And IMO you should consider doing the same instead of adding arbitrary "workarounds" (aka hacks).